ui/packages/consul-ui/app/serializers/session.js

```diff
@@ -7,7 +7,20 @@ export default class SessionSerializer extends Serializer {
 
   respondForQueryRecord(respond, query) {
     return super.respondForQueryRecord(
-      cb => respond((headers, body) => cb(headers, body[0])),
+      cb =>
+        respond((headers, body) => {
+          if (body.length === 0) {
+            const e = new Error();
+            e.errors = [
+              {
+                status: '404',
+                title: 'Not found',
+              },
+            ];
+            throw e;
+          }
+          return cb(headers, body[0]);
+        }),
```

Comment:
When requesting a single session, I noticed that this is one of those Consul "it doesn't exist but I'll give you a 200 with an empty array" things. We now check for that and throw a fake 404. We used to cope with this differently but this makes more sense. Another little FYI is that we have a custom `HTTPError` shaped `Error` we could use for all of these at some point.

Reply:
> I'll give you a 200 with an empty array" things

There's a perfectly good HTTP status code for this (I realize this is not the place to take my complaints haha) 😭 😭 😭

Reply:
🤣 yeah we have a few of these that date back quite a few years.

What is meant here is "The session you requested is not found" but Consul gives us a `[]` because here it always responds with arrays even for requests for single sessions. I think this is one of the cases where we don't change things with the benefit of hindsight for back-compat reasons, but no reason why we can't fake a 404 in the UI data layer.